app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -0,0 +1,405 @@
+package org.oppia.android.app.spotlight
+
+import android.content.Context
+import android.content.res.Resources
+import android.util.DisplayMetrics
+import android.util.Log
+import android.view.View
+import android.view.ViewGroup
+import android.view.animation.DecelerateInterpolator
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import com.takusemba.spotlight.OnSpotlightListener
+import com.takusemba.spotlight.OnTargetListener
+import com.takusemba.spotlight.Spotlight
+import com.takusemba.spotlight.Target
+import com.takusemba.spotlight.shape.Circle
+import com.takusemba.spotlight.shape.RoundedRectangle
+import com.takusemba.spotlight.shape.Shape
+import org.oppia.android.R
+import org.oppia.android.app.fragment.FragmentComponentImpl
+import org.oppia.android.app.fragment.InjectableFragment
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.app.model.SpotlightViewState
+import org.oppia.android.app.onboarding.SpotlightNavigationListener
+import org.oppia.android.app.topic.PROFILE_ID_ARGUMENT_KEY
+import org.oppia.android.databinding.BottomLeftOverlayBinding
+import org.oppia.android.databinding.BottomRightOverlayBinding
+import org.oppia.android.databinding.TopLeftOverlayBinding
+import org.oppia.android.databinding.TopRightOverlayBinding
+import org.oppia.android.domain.spotlight.SpotlightStateController
+import org.oppia.android.util.accessibility.AccessibilityServiceImpl
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import java.util.LinkedList
+import java.util.Locale
+import javax.inject.Inject
+
+class SpotlightFragment : InjectableFragment(), SpotlightNavigationListener {
+  @Inject
+  lateinit var activity: AppCompatActivity
+
+  @Inject
+  lateinit var spotlightStateController: SpotlightStateController
+
+  @Inject
+  lateinit var accessibilityServiceImpl: AccessibilityServiceImpl
+
+  private var targetList = LinkedList<Target>()
```

Comment:
As mentioned during our meeting, prefer using a mutable list, instead, for a slightly more Kotlin-esque approach.